src/common/SimulationFloorInputData.java

```diff
@@ -0,0 +1,127 @@
+/**
+ *
+ */
+package common;
+
+import java.security.InvalidParameterException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * This class stores the properties of a given simulation floor input.
+ *
+ * @author paulokenne
+ *
+ */
+public class SimulationFloorInputData {
+
+	/**
+	 * The floor data input separator
+	 */
+	private static final String FLOOR_DATA_INPUT_SEPARATOR = " ";
+
+	/**
+	 * The data formatter for the time field
+	 */
+	private final SimpleDateFormat DATA_FORMAT_FOR_ARRIVAL_TIME = new SimpleDateFormat("hh:mm:ss.mmm");
+	/**
+	 * The time stamp indicating when the passenger arrives.
+	 */
+	private String arrivalTime;
+
+	/**
+	 * The current floor.
+	 */
+	private Integer currentFloor;
+
+	/**
+	 * The elevator direction that the passenger wishes to go to reach his or her
+	 * destination
+	 */
+	Direction floorDirectionButton;
+
+	/**
+	 * The floor that the passenger wishes to go to.
+	 **/
+	private Integer destinationFloorCarButton;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param arrivalTime               the arrival time
+	 * @param currentFloor              the current floor
+	 * @param floorDirectionButton      the floor direction
+	 * @param destinationFloorCarButton the target floor
+	 */
+	public SimulationFloorInputData(String arrivalTime, Integer currentFloor, Direction floorDirectionButton,
+			Integer destinationFloorCarButton) {
+		this.arrivalTime = arrivalTime;
+		this.currentFloor = currentFloor;
+		this.floorDirectionButton = floorDirectionButton;
+		this.destinationFloorCarButton = destinationFloorCarButton;
+	}
+
+	/**
+	 * Constructor.
+	 *
+	 * @param dataString the data line string with formating: "Time Floor
+	 *                   FloorButton CarButton"
+	 */
+	public SimulationFloorInputData(String dataString) throws InvalidParameterException {
+
+		try {
+			// Format of data String: Time Floor FloorButton CarButton
+			String[] data = dataString.split(FLOOR_DATA_INPUT_SEPARATOR);
+
+			System.out.println(data.length);
+			Date parsedDate = DATA_FORMAT_FOR_ARRIVAL_TIME.parse(data[0]);
+			this.arrivalTime = DATA_FORMAT_FOR_ARRIVAL_TIME.format(parsedDate);
+
+			this.currentFloor = Integer.parseInt(data[1]);
+			this.floorDirectionButton = Direction.valueOf(data[2]);
+			this.destinationFloorCarButton = Integer.parseInt(data[3]);
+
+		} catch (Exception e) {
+			System.out.println(e);
+			throw new InvalidParameterException(dataString);
+		}
+
+	}
+
+	/**
+	 * Gets the arrival time
+	 *
+	 * @return the arrivalTime
+	 */
+	public String getArrivalTime() {
+		return arrivalTime;
+	}
```

Comment:
Could probably remove these comments on the getters. They just serve as clutter.

Reply:
I feel you, bro. The only reason I added them is for scalability purposes. 

For example, we may want to do things to properties before setting or getting. Even if the project is large,  such a change would be simple: modify the getter and setter functions.

If we do not have getter and setter functions, that change would be a menace. We would have to find every instance that access the property using _data.property_. After finding them, we would have to replace every instance which is not fun.